courses/serializers/v1/courses.py

```diff
@@ -33,7 +33,9 @@ def get_programs(self, instance):
         if self.context.get("all_runs", False):
             from courses.serializers.v1.base import BaseProgramSerializer
 
-            return BaseProgramSerializer(instance.programs, many=True).data
+            return BaseProgramSerializer(
+                [program for program in instance.programs if program.live], many=True
+            ).data
```

Comment:
So, I think this also needs to check page__live - I'm adding an endpoint call for that now actually (similar behavior on catalog), but it should also check if program.page.live

Reply:
Otherwise a program which is live but does not yet have a page will be linked (and said link will be broken)

Reply:
Should be fixed in https://github.com/mitodl/mitxonline/pull/1989/commits/eb53287296b28a682c923bea12f58e689e603042

Reply:
yeah :)  I don't see the program I tested with any more!

There's a test failing now, but it's passing locally so I'm not sure. The data on that test makes sense on first pass, but I didn't think too long on it.